qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/IndexingIT.java

```diff
@@ -5,28 +5,181 @@
 
 package org.opensearch.knn.bwc;
 
+import org.opensearch.common.Strings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.knn.index.SpaceType;
+
+import java.util.Optional;
+
+import static org.opensearch.knn.TestUtils.KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE;
+import static org.opensearch.knn.TestUtils.KNN_ALGO_PARAM_M_MIN_VALUE;
+import static org.opensearch.knn.TestUtils.KNN_ENGINE_FAISS;
 import static org.opensearch.knn.TestUtils.NODES_BWC_CLUSTER;
+import static org.opensearch.knn.common.KNNConstants.KNN_METHOD;
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+import static org.opensearch.knn.common.KNNConstants.NAME;
+import static org.opensearch.knn.common.KNNConstants.PARAMETERS;
 
 public class IndexingIT extends AbstractRestartUpgradeTestCase {
     private static final String TEST_FIELD = "test-field";
     private static final int DIMENSIONS = 5;
     private static final int K = 5;
-    private static final int ADD_DOCS_CNT = 10;
+    private static final int ADD_DOCS_COUNT = 10;
 
-    public void testKnnDefaultIndexSettings() throws Exception {
+    // Default Legacy Field Mapping
+    // space_type : "l2", engine : "nmslib", m : 16, ef_construction : 512
+    public void testKnnIndexDefaultLegacyFieldMapping() throws Exception {
         waitForClusterHealthGreen(NODES_BWC_CLUSTER);
 
         if (isRunningAgainstOldCluster()) {
             createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
-            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_CNT);
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_COUNT);
+        }
+
+        else {
+            kNNIndexingTestsUpgradedCluster();
+        }
+    }
+
+    // Custom Legacy Field Mapping
+    // space_type : "linf", engine : "nmslib", m : 2, ef_construction : 2
+    public void testKnnIndexCustomLegacyFieldMapping() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(
+                testIndex,
+                getKNNIndexCustomLegacyFieldMappingSettings(
+                    SpaceType.LINF,
+                    KNN_ALGO_PARAM_M_MIN_VALUE,
+                    KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE
+                ),
+                createKnnIndexMapping(TEST_FIELD, DIMENSIONS)
```

Comment:
are you testing anything inside if block? Looks like set up block.

Reply:
Yes, for this test it is more of like a setup where we will be creating an index with custom mapping and add documents to it.

Reply:
Can you please add this as comment?

Reply:
added